File: src/main/java/com/mellivorines/generator/entity/GenFieldType.java
package com.mellivorines.generator.entity;


import org.babyfish.jimmer.sql.Column;
import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.GeneratedValue;
import org.babyfish.jimmer.sql.GenerationType;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.Table;
import org.jetbrains.annotations.Nullable;

@Entity
@Table(name = "gen_field_type")
public interface GenFieldType {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    int id();

    @Nullable
    @Column(name = "column_type")
    String columnType();

    @Nullable
    @Column(name = "attr_type")
    String attrType();

    @Nullable
    @Column(name = "package_name")
    String packageName();

}

File: src/main/kotlin/com/mellivorines/generator/entity/GenFieldType.kt
package com.mellivorines.generator.entity

import org.babyfish.jimmer.sql.Column
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.GeneratedValue
import org.babyfish.jimmer.sql.GenerationType
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.Table

@Entity
@Table(name = "gen_field_type")
interface GenFieldType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    @Column(name = "column_type")
    val columnType: String?

    @Column(name = "attr_type")
    val attrType: String?

    @Column(name = "package_name")
    val packageName: String?
}